content/build.ts

```diff
@@ -0,0 +1,500 @@
+/**
+ * Build Script for Navigator Content
+ *
+ * This script generates static JSON files from content sources (markdown, JSON) for use
+ * in the business.nj.gov My Account application. It processes 8 content types:
+ * industries, tasks, actions, certifications, filings, fundings, license events, and reinstatements.
+ *
+ * ## Architecture
+ *
+ * 1. **Infrastructure Layer** - File system operations (readFile, writeFile, etc.)
+ *    - Abstracted behind FileSystemPort interface for dependency injection
+ *    - Makes the code 100% testable without touching the real file system
+ *
+ * 2. **Domain Layer** - Pure business logic (parseIndustryFile, buildIndustries, etc.)
+ *    - No dependencies on file system or external libraries
+ *    - Receives file system port as parameter for all I/O operations
+ *
+ * 3. **Application Layer** - Orchestration (buildAndWriteContent, executeBuild, main)
+ *    - Coordinates domain functions with infrastructure
+ *    - Uses metaprogramming pattern (contentConfigs array) to eliminate code duplication
+ *
+ * ## Metaprogramming Pattern
+ *
+ * Instead of writing 7 nearly-identical build functions for each of the Markdown imports, we use a configuration-driven approach:
+ * - contentConfigs array defines metadata for each content type
+ * - buildAndWriteContent() generic helper handles all content types
+ * - executeBuild() loops through configs instead of calling individual functions
+ */
```

Comment:
_Since_ the build script is a standalone file, no one will object to me adding JSDoc as a pattern into the codebase right?

I started out by keeping the build script as .js but to do type checking using JSDoc comments, but it was unnecessarily complex of an approach and we don't do that anywhere else in our codebase.

Plus TS -> TS imports became easier, and I'd rather not fight our build system if I don't have to.